src/Initializer/InputAux.hpp

```diff
@@ -103,6 +103,59 @@ void convertStringToMask(const std::string& stringMask, ContainerT& mask) {
   }
 }
 
+/**
+ * \brief Converts an input string to a vector of the datatype T. If the input string does not contain enough
+ * items to convert to an array of length n, we fill with default values. If the input string contains
+ * more than n items, we only consider the first n ones.
+ *
+ * \throws runtime_error if the input string contains parameters, which can not be converted to T.
+ * */
+template<typename T, size_t n>
+std::array<T, n> convertStringToArray(const std::string& inputString) {
```

Comment:
I don't really like that you return a default initialized array when the string is empty/has fewer entries. This could be very confusing. Maybe add another parameter defaultValue and fill the array manually?

Reply:
function should maybe be also be called splitAndConvert? Maybe make delimiter configurable in the future

Reply:
Ok, we now do a "hard fail" by default.